### PR TITLE
feat(services): migrate service commits to react query

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-891

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Make it work, make it right, make it fast

File: libs/shared/util-types/src/lib/shared-util-types.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+type UnionTypeGuard<T> = T extends (o: any) => o is infer U ? U : never
+
+export function unionTypeGuard<T extends GenericTypeGuard>(guards: T[]): (x: any) => x is UnionTypeGuard<T> {
+  return (value): value is UnionTypeGuard<T> => guards.some((guard) => guard(value))
+}
+
+type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (k: infer I) => void ? I : never
+type IntersectionTypeGuard<T> = UnionToIntersection<UnionTypeGuard<T>>
+
+export function intersectionTypeGuard<T extends GenericTypeGuard>(
```
Comment (RemiBonnet) [inline libs/shared/util-types/src/lib/shared-util-types.ts:13]: You don't use it ?

**Review by RemiBonnet — APPROVED** [review]

File: libs/shared/util-types/src/lib/shared-util-types.ts (hunk 1 above)
Comment (ctjhoa) [reply libs/shared/util-types/src/lib/shared-util-types.ts:13]: Not for now but this can be useful in the future

**Pull request merged**